src/util.js

```diff
@@ -65,45 +50,42 @@ function serialize (node, callback) {
     })
   }
 
-  try {
-    serialized = proto.PBNode.encode(toProtoBuf({
-      data, links
-    }))
-  } catch (err) {
-    return callback(err)
-  }
+  const serialized = proto.PBNode.encode(toProtoBuf({
+    Data: data,
+    Links: links
+  }))
 
-  callback(null, serialized)
+  return Uint8Array.from(serialized)
 }
 
-function deserialize (buffer, callback) {
-  const pbn = proto.PBNode.decode(buffer)
+const deserialize = async (buffer) => {
+  const pbn = proto.PBNode.decode(Buffer.from(buffer))
```

Comment:
Previously we didn't copy this buffer before decoding it - did that cause a problem that this copy solves?

Reply:
`protons` is [one of ours](https://github.com/ipfs/protons), can we not just fix it there?

Reply:
That's the reason, if we use https://github.com/mapbox/pbf we get a faster thing and don't have to maintain it!

Reply:
The problem is that bubbling up the changes up to js-ipfs is a huge pain, it takes weeks. Hence I'd like to do as little breaking releases as possible. Though I'll think about that as keeping it would result in less per regressions and I could make a clean PR which is exchanging the PB lib and changes the return value to Uint8Array.

Reply:
> I could make a clean PR which is exchanging the PB lib and changes the return value to Uint8Array

I think this would be preferable

> Bubbling up the changes up to js-ipfs is a huge pain, it takes weeks

Agreed. I think maybe can coordinate better as a team - maybe have a week where will all try to coordinate and ship the same feature - be that base32 CIDs or Uint8Arrays, etc.  Let's bring it up at next week's js-core catchup.